app/services/email_generation_service.rb

```diff
@@ -7,41 +7,39 @@ def self.call(*args)
 
   def call
     SubscriptionContent.with_advisory_lock(LOCK_NAME, timeout_seconds: 0) do
```

Comment:
Minor suggestion: it might be nice to have a method name to make it clearer what this does, e.g.

```ruby
only_run_this_code_if_no_other_process_is_running_this_code do
  # ...
end
```

(hopefully with a better name!)

What do you think?

Reply:
`ensure_only_running_once` ?